spylon_kernel/scala_magic.py

```diff
@@ -78,28 +87,70 @@ def _loop_alive(self):
             yield gen.sleep(0.01)
 
     def line_scala(self, *args):
-        """
-        %scala CODE - evaluate code as Scala
-        This line magic will evaluate the CODE (either expression or
-        statement) as Scala code.
-
-        Examples:
-            %scala val x = 42
-            %scala import scala.math
-            %scala x + math.pi
+        """%scala CODE - evaluates a line of code as Scala
+
+        Parameters
+        ----------
+        *args : list of string
+            Line magic arguments joined into a single-space separated string
+
+        Examples
+        --------
+        %scala val x = 42
+        %scala import scala.math
+        %scala x + math.pi
         """
         code = " ".join(args)
         self.eval(code, True)
 
+    @option(
+        "-e", "--eval_output", action="store_true", default=False,
+        help="Evaluate the return value from the Scala cell as code in the kernel language."
+    )
+    def cell_scala(self, eval_output=False):
+        """%%scala - evaluate contents of cell as Scala code
+
+        This cell magic will evaluate the cell (either expression or statement) as
+        Scala code. This will instantiate a scala interpreter prior to running the code.
+
+        The -e or --eval_output flag signals that the retval value expression will
+        be used as code for the cell to be evaluated by the host language.
+
+        Examples
+        --------
+        %%scala
+        val x = 42
+
+        %%scala
+        import collections.mutable._
+        val y = mutable.Map.empty[Int, String]
+
+        %%scala -e
+        retval = "'(this is code in the kernel language)"
+
+        %%python -e
+        "'(this is code in the kernel language)"
+        """
+        if self.code.strip():
```

Comment:
add a note to this if block. I'm guessing this is to skip attempting to evaluate an empty cell. Future everyone appreciates this context